samza-yarn/src/main/java/org/apache/samza/job/yarn/FsImplConfigManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.job.yarn;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.MapConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * FsImplConfigManager is intented to manage the Samza config for fs.&lt;scheme&gt;impl.
+ * e.g. fs.http.impl
+ *
+ * Usually this config exist in yarn config,
+ * but it is also allowed the clients define this for Samza Job and pick specific implementation for the job.
+ */
+public class FsImplConfigManager {
```

Comment:
I looked at several cases like this in our code base, like JavaStorageConfig, etc. But it used inherence instead of composition. From my perspective, I think it is better to use composition over inherence for this case, as it is easier to focus only the business logic for the specific config. If we use inheritance, we also expose the get() function which can return any config values although a lot of them are not related to this specific config type, for example, JavaStorageConfig can even return config not related to storage, which does not make too much sense to me. 

I propose to use composition for this case and do migration for other cases. But if there is a strong preferences for doing inherence for this case, I am OK to change it back to inherence as well.